src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -119,12 +107,20 @@ public class NodeSettings
                 this.DataDir = Directory.CreateDirectory(directoryPath).FullName;
                 this.Logger.LogDebug("Data directory initialized with path {0}.", this.DataDir);
             }
-
+            
             this.DataFolder = new DataFolder(this.DataDir);
 
-            // Load configuration from .ctor?
-            if (loadConfiguration)
-                this.LoadConfiguration();
+            // Can determine the configuration file name now if it was not specified on the command line.
+            if (this.ConfigurationFile == null)
+            {
+                this.ConfigurationFile = Path.Combine(this.DataDir, this.Network.DefaultConfigFilename);
+                this.Logger.LogDebug("Configuration file set to '{0}'.", this.ConfigurationFile);
+
+                this.SetCombinedConfiguration();
```

Comment:
SetCombinedConfiguration ends up being called twice if we don't have a config file, which I find quite confusing. 
Also, previously there was 1 place where we called the MergeInto method, now I count SetCombinedConfiguration being called in 3 different places.

Reply:
In the past we had a LoadConfiguration call in both NodeSettings and the FullNodeBuilder, plus we read the config (temporarily) to get at the "testnet" and "regtest" settings in the constructor. So technically we were already loading the configuration three times.

This approach makes the access to the configuration more consistent - as we always access it the same way.